src/CoreWCF.Primitives/src/CoreWCF/Channels/MessageEncoderCompressionHandler.cs

```diff
@@ -13,45 +17,30 @@ internal static class MessageEncoderCompressionHandler
         internal const string DeflateContentEncoding = "deflate";
         private const int DecompressBlockSize = 1024;
 
-        internal static void DecompressBuffer(ref ArraySegment<byte> buffer, BufferManager bufferManager, CompressionFormat compressionFormat, long maxReceivedMessageSize)
+        internal static async ValueTask<ReadOnlySequence<byte>> DecompressBufferAsync(ReadOnlySequence<byte> buffer, CompressionFormat compressionFormat, long maxReceivedMessageSize)
         {
```

Comment:
No need to make the method async as the source data is coming from a buffer. You can call the synchronous Stream.CopyTo() and not need anything async.

Reply:
done